src/@next/Select/components/SearchableSelectInput/SearchableSelectInput.tsx

```diff
@@ -140,10 +140,20 @@ export const SearchableSelectInput = forwardRef<
   useEffect(() => {
     if (selectedValue) {
       setShowClear(true);
+      updateSearchableSelectState({
+        showSelected: true,
+        showInput: false,
+        showPlaceholder: false,
+      });
       return;
     }
 
     setShowClear(false);
+    updateSearchableSelectState({
+      showSelected: false,
+      showInput: true,
+      showPlaceholder: true,
+    });
   }, [selectedValue]);
```

Comment:
@farhanabi there's seem to be a warning about useEffect dependencies here

Reply:
Any idea how to solve this? If I add `updateSearchableSelectState` to the dependency array, it will do a lot of unnecessary render. I tried wrapping `updateSearchableSelectState` with `useCallback` in the parent component when it initializes, the rendering keeps happening.

Reply:
Does it need to be run under useEffect? The function seems to depend on `selectedValue` or `showClear`?